CollAction/Startup.cs

```diff
@@ -127,9 +128,74 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
+                    .AddFrameOptionsDeny() // No framing allowed (put us inside a frame tag)
+                    .AddContentSecurityPolicy(cspBuilder =>
+                    {
+                        cspBuilder.AddBlockAllMixedContent(); // Block mixed http/https content
+                        cspBuilder.AddUpgradeInsecureRequests(); // Upgrade all http requests to https
+                        cspBuilder.AddObjectSrc().Self(); // Only allow plugins/objects from our own site
+                        cspBuilder.AddFormAction().Self(); // Only allow form actions to our own site
+                        cspBuilder.AddConnectSrc().Self() // Only allow API calls to self, and the websites we use for the share buttons
+                                                  .Sources.AddRange(new[]
+                                                                    {
+                                                                        "https://www.linkedin.com/",
+                                                                        "https://linkedin.com/",
+                                                                        "https://www.twitter.com/",
+                                                                        "https://twitter.com/",
+                                                                        "https://www.facebook.com/",
+                                                                        "https://facebook.com/",
+                                                                        "https://graph.facebook.com/"
+                                                                    });
+                        cspBuilder.AddImgSrc().Self() // Only allow self-hosted images, or google analytics (for tracking images)
+                                              .Sources.AddRange(new[]
+                                                                {
+                                                                    "https://www.google-analytics.com"
+                                                                });
+                        cspBuilder.AddStyleSrc().Self() // Only allow style/css from these sources (note: css injection can actually be dangerous)
+                                                .UnsafeInline() // Unfortunately this is necessary, the backend passess some things that are directly passed into the css, especially on the project page. TODO: We should try to get rid of this.
+                                                .Sources.AddRange(new[]
+                                                                  {
+                                                                      "https://maxcdn.bootstrapcdn.com/",
```

Comment:
Could think of some way to dynamically retrieve style/font sources via provider, at this time we must ensure to remember to add to here if some new source is used in frontend.

Reply:
Problem is that currently the fonts are retrieved through some 3rd party javascript, which then loads things from another domain. Don't see an easy feasible way to do it. If you have a good refactoring, feel free to add.